taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,124 @@ task_history(taskId : number) : Promise<Object> {
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid: false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
+    //which is  ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    if((new_entry.status_trans[0]!="New"&&new_entry.status_trans[1]!="In progress")
+        &&(new_entry.status_trans[0]!="In progress"&&new_entry.status_trans[1]!="Ready for test")
+        &&(new_entry.status_trans[0]!="Ready for test"&&new_entry.status_trans[1]!="Closed"))
+        new_entry.state_trans_invalid= true;
+    output.push(new_entry);
+}
+
+return output;
+}
+
+
+**
+ * @summary This call return a task current end status and estimated finished percentage based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task is abnormal,task status transition, date based on task Id
+ * * {
+ * Current in Use
+ *      finished: Boolean           // true, the task is finished.
+ *      fin_per: number            // finished percentage % default [0%,10%,60%,100%]
+ *      end_status: string,       // current task status  ["New", "In progress","Ready for test", "Closed"]
+ *      num_stat:number           // current task status [0 1 2 3] corresponding to string in end_status
+ *
+ * }
+ */
+export async function
+task_assessment_endstate(taskId : number) : Promise<Object> {
+
+    let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741   or 2555550
+    let output : Array<Object> = [];
+    let endstatus:string = "New";
+    for(let entry of data) {
+        let new_entry = {
+            state_trans_invalid: false,
+            date : new Date(entry.created_at).getTime(),
+            user : entry.user,
+            status_trans:entry.values_diff.status
+        }
+        endstatus=new_entry.status_trans[1];
+        output.push(new_entry);
+    }
+
+    //endstatus="Ready for test";//For test use
+    let  t_finished: boolean;
+    let t_fin_per:number;
+    let t_num_stat:number;
+     if(endstatus!="Closed") {
+         t_finished = false;
+     }
+    else{
+        t_finished = true;
+    }
+
+
+    //transfer ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    //into number status
+    switch(endstatus){
+        case "Closed":{
+            t_fin_per = 100;
+            t_num_stat = 3;
+            break
+        }
+        case "Ready for test":{
+            t_fin_per = 60;
+            t_num_stat = 2;
+            break
+        }
+        case "In progress":{
+            t_fin_per = 10;
+            t_num_stat = 1;
+            break
+        }
+        case "New":{
```

Comment:
why does a "new" case have a higher finished % (I think? Your variable names are a little hard to follow) than in progress?
Do you have research to back up these percentage distributions? (It's okay if you don't right now, I just think we should not have magic numbers dictating analysis rather some real world backing on how to assess this so please consider this for the future of your work)

Reply:
to build on this. I don't understand the correlation between t_fin_per and t_num_stat nor understand how this will convey value to a potential user. I feel a comment explaining this should exist in the code whether to a link or a brief summary. If there is no reasonable correlation then there should be work to synthesize one.

Reply:
fix new case, that is a typo.   t_num_stat  will translate state "New", "In progress",etc into number so I can be easily plotted.  percentage distributions is used to have a inspect of  a sprint while it is going on, and provide a additional view of burndown chart. The numbers I picked is from statistical data. For users using our tool, they can change the estimated percentage based on their experience.